implementations/elixir/ockam/ockam/lib/ockam/secure_channel/channel.ex

```diff
@@ -0,0 +1,133 @@
+defmodule Ockam.SecureChannel.Channel do
+  @moduledoc false
+
+  use GenStateMachine
+
+  alias Ockam.Node
+  alias Ockam.SecureChannel.KeyEstablishmentProtocol.XX, as: XXKeyEstablishmentProtocol
+  alias Ockam.Telemetry
+
+  @doc false
+  def send(channel, message), do: Node.send(channel, message)
+
+  @doc false
+  def peer(_channel), do: :ok
+
+  @doc false
+  def create(options) when is_list(options) do
+    options = Keyword.put_new_lazy(options, :address, &Node.get_random_unregistered_address/0)
+
+    case Node.start_supervised(__MODULE__, options) do
+      {:ok, _pid, address} -> {:ok, address}
+      error -> error
+    end
+  end
+
+  @doc false
+  def start_link(options) when is_list(options) do
+    with {:ok, address} <- get_from_options(:address, options),
+         {:ok, pid} <- start(address, options) do
+      {:ok, pid, address}
+    end
+  end
+
+  defp start(address, options) do
+    name = {:via, Node.process_registry(), address}
+    GenStateMachine.start_link(__MODULE__, options, name: name)
+  end
+
+  @doc false
+  @impl true
+  def init(options) do
+    metadata = %{options: options}
+    start_time = Telemetry.emit_start_event([__MODULE__, :init], metadata: metadata)
+
+    with {:ok, data} <- setup_plaintext_address(options, %{}),
+         {:ok, data} <- setup_ciphertext_address(options, data),
+         {:ok, data} <- setup_route_to_peer(options, data),
+         {:ok, data} <- setup_initiating_message(options, data),
+         {:ok, initial_state, data} <- setup_key_establisment_protocol(options, data) do
+      return_value = {:ok, initial_state, data}
+
+      metadata = Map.put(metadata, :return_value, return_value)
+      Telemetry.emit_stop_event([__MODULE__, :init], start_time, metadata: metadata)
+
+      return_value
+    end
+  end
+
+  @doc false
+  @impl true
+  def handle_event(event_type, event, state, data) do
+    metadata = %{event_type: event_type, event: event, state: state, data: data}
+    start_time = Telemetry.emit_start_event([__MODULE__, :handle_event], metadata: metadata)
+
+    return_value = handle_message(event_type, event, state, data)
+
+    metadata = Map.put(metadata, :return_value, return_value)
+    Telemetry.emit_stop_event([__MODULE__, :handle_event], start_time, metadata: metadata)
+
+    return_value
+  end
+
+  defp handle_message(:info, event, {:key_establishment, _role, _role_state} = state, data) do
+    key_establisment_protocol = Map.get(data, :key_establisment_protocol)
+    key_establisment_protocol.handle_message(event, state, data)
```

Comment:
typo: `s/establisment/establishment`

Reply:
thanks, will fix

Reply:
ignore any repetitions of this ... I'll search and replace